### PR TITLE
HIP Documentation Refactor to incorporate new diataxis based organization and collecting other stale doc PRs

### PR DESCRIPTION
**Pull request by LisaDelaney**

### CODE REVIEW

File: docs/how-to/logging.rst [hunk 1]
```diff
@@ -0,0 +1,235 @@
+.. meta::
+   :description: HIP provides a logging mechanism that allows you to trace HIP API and runtime codes
+   when running a HIP application.
+   :keywords: AMD, ROCm, HIP, logging
```
Comment (saadrahim) [inline docs/how-to/logging.rst:4]: @samjwu  can you help deal with common keywords centrally?

File: docs/how-to/debugging.rst [hunk 2]
```diff
@@ -0,0 +1,381 @@
+.. meta::
+   :description: How to debug using HIP.
+   :keywords: AMD, ROCm, HIP, debugging, ltrace, ROCdgb, Windgb
+
+*************************************************************************
+Debugging with HIP
+*************************************************************************
```
Comment (saadrahim) [inline docs/how-to/debugging.rst:7]: I suspect this entire section is too old.

File: docs/how-to/debugging.rst [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    PASSED!
+
+Debugging
+================================================
+
+You can use ROCgdb for debugging and profiling.
+
+ROCgdb is the ROCm source-level debugger for Linux and is based on GNU Project debugger (GDB).
+the GNU source-level debugger, equivalent of cuda-gdb, can be used with debugger frontends, such as eclipse, vscode, or gdb-dashboard.
+For details, see (https://github.com/ROCm-Developer-Tools/ROCgdb).
```
Comment (saadrahim) [inline docs/how-to/debugging.rst]: link via intersphinx to rocGDB docs, not github page.

File: docs/how-to/debugging.rst [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+    ``HSA_ENABLE_SDMA=0`` causes host-to-device and device-to-host copies to use compute shader
+    blit kernels, rather than the dedicated DMA copy engines. Compute shader copies have low latency
+    (typically < 5 us) and can achieve approximately 80% of the bandwidth of the DMA copy engine.
+
+* To diagnose interrupt storm issues in the driver, you can use ``HSA_ENABLE_INTERRUPT``.
+
+    ``HSA_ENABLE_INTERRUPT=0`` causes completion signals to be detected with memory-based
+    polling, rather than interrupts.
+
+HIP environment variable summary
```
Comment (saadrahim) [inline docs/how-to/debugging.rst:270]: I'd like to environment variables into new page and link.

File: docs/install/install.rst [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+   .. tab-item:: AMD
+      :sync: amd
+
+      HIP is automatically installed during the ROCm installation. If you haven't yet installed ROCm, you
+      can find installation instructions here:
+
+         * :doc:`rocm:/install/linux/install`
+         * :doc:`rocm:/install/windows/install`
+
+By default, HIP is installed into /opt/rocm/hip.
```
Comment (saadrahim) [inline docs/install/install.rst:44]: No longer accurate, hip is not installed into this directory.

File: docs/install/install.rst [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+            .. code:: shell
+
+               make directed_tests.texture.hipTexObjPitch
+               cd $CLR_DIR/build/hipamd/directed_tests/texture
+               ./hipTexObjPitch
+
+         .. note::
+            The integrated HIP directed tests will be deprecated in a future release.
+
+      * Build HIP catch tests.
```
Comment (saadrahim) [inline docs/install/install.rst:267]: We should not call it catch tests, talk to HIP team. They are probably unit tests, functional tests, etc.

File: docs/old/reference/kernel_language.rst [hunk 7]
```diff
@@ -0,0 +1,2050 @@
...
+
+The 32-bit int predicate from each lane reduces to a 1-bit value of 0 (predicate = 0) or 1
+(predicate != 0). To get a summary view of the predicates that are contributed by other warp lanes, you
+can use:
+
+* ``__any()``: Returns 1 if any warp lane contributes a nonzero predicate, otherwise it returns 0
+* ``__all()``: Returns 1 if all other warp lanes contribute nonzero predicates, otherwise it returns 0
+
+To determine if the target platform supports the any/all instruction, you can use the ``hasWarpVote``
+device property or the ``HIP_ARCH_HAS_WARP_VOTE`` compiler definition.
```
Comment (saadrahim) [inline docs/old/reference/kernel_language.rst]:
```suggestion
device property or the ``HIP_ARCH_HAS_WARP_VOTE`` header file macro definition.
```
I think this is the more formal term, I would consult with @MathiasMagnus or @gargrahul .

https://github.com/ROCm-Developer-Tools/HIP/blob/481fc5fcec1bb2a49b17a1b96454bc8b019329e0/include/hip/hip_common.h#L81

File: docs/old/reference/kernel_language.rst [hunk 8]
```diff
@@ -0,0 +1,2050 @@
...
+```
+[TARGET GPU] = GPU architecture
+[INPUT FILE] = Name of the file containing kernels
+[OUTPUT FILE] = Name of the generated code object file
+```
+
+Note: When using binary code objects is that the number of arguments to the kernel is different on HIP-Clang and NVCC path. Refer to the sample in samples/0_Intro/module_api for differences in the arguments to be passed to the kernel.
+
+## gfx-arch-specific-kernel
+Clang defined '__gfx*__' macros can be used to execute gfx arch specific codes inside the kernel. Refer to the sample 14_gpu_arch in samples/2_Cookbook.
```
Comment (saadrahim) [inline docs/old/reference/kernel_language.rst]: @MathiasMagnus I think we have newer examples available for this now. is that right?

File: docs/how-to/logging.rst (hunk 1 above)
Comment (samjwu) [reply docs/how-to/logging.rst:4]: as in always insert AMD and ROCm along with a meta directive?

Comment (saadrahim) [reply docs/how-to/logging.rst:4]: Yes, AMD, ROCm and the project name. Let's start with those three

**Comment by chrispaquot** [comment]
Is this change still needed?

**Comment by jujiang-del** [comment]
Changes in PR#3432 have been merged in Gerrit.
Does PR#3432 cover all the changes in this PR?
If so, please close this PR.
Thanks.

**Comment by mangupta** [comment]
fee13d6a3f2d925464fa22fb45e752cb014f8017 contains the change. Please raise a new PR in case you find something still missing.

**Pull request closed without merging**